### ISSUE TITLE
Soft descriptor missing for PayPalExpress

### ISSUE THREAD

**Issue by talyssonoc** [activemerchant/active_merchant#2152]
Is it possible to add descriptors to the purchase using `PayPalExpress`? I noticed it's possible on [paypal.rb](https://github.com/activemerchant/active_merchant/blob/v1.59.0/lib/active_merchant/billing/gateways/paypal.rb#L77) but not on [paypal_express.rb](https://github.com/activemerchant/active_merchant/blob/v1.59.0/lib/active_merchant/billing/gateways/paypal_express.rb#L108). Somehow it was working when I was using v1.44.1 (I wasn't able to discover what was making it work), so it may be related to the fact I've updated to v1.59.0. 

After I complete a purchase, if I go to the purchase page on PayPal dashboard, it won't show the soft descriptor I've send neither on `PayPalExpressGateway#setup_purchase` or `PayPalExpressGateway#purchase`.

ActiveMerchant version: 1.59.0
Ruby version: 2.1.6
ActiveSupport version: 4.1.9
Nokogiri version: 1.6.8

Could you guys help me?

**Comment by talyssonoc** [comment]
I've found out that it's missing a `n2:SoftDescriptor` in that array: https://github.com/activemerchant/active_merchant/blob/v1.60.0/lib/active_merchant/billing/gateways/paypal/paypal_common_api.rb#L623. If it's the correct fix, I'll be glad to send a PR to fix that.

/cc @duff @ntalbott @Soleone @melari

**Comment by talyssonoc** [comment]
@Soleone I believe it was removed here: https://github.com/activemerchant/active_merchant/commit/aec39c6fb89ed335c43ef284db3d7e382ede1bd9#diff-ab5a6a27fa046b2655fb1cae4c373356L607

In this commit it was moved to a different place, but then in this other commit (https://github.com/activemerchant/active_merchant/commit/e7b35c2ea9925dbecf6424e86353ae1e9daad794#diff-ab5a6a27fa046b2655fb1cae4c373356L584) it was moved to the `pay_pal.rb` file, that PayPal Express gateway doesn't use. I think it may be a good reason to put it back on `paypal_common_api.rb` again?

**Comment by duff** [comment]
@talyssonoc Thanks for the PR!

I'm sorry I broke Paypal Express. 😭 Looking back over the commit, I'm unable to discern why I moved that logic from the paypal_common to the child class.  I regret not adding more detail in that commit message.

**Comment by talyssonoc** [comment]
@duff I'm glad to help!

**Comment by wyncg** [comment]
This thread is pretty old but recently I faced similar issue. I am using `PAYPAL_GATEWAY.reference_transaction(price,` options)`, and in `options` I set

**Comment by wyncg** [comment]
This thread is pretty old but recently I faced similar issue. I am using
`PAYPAL_GATEWAY.reference_transaction(price, options)`
and in `options` I set:
```
options[:soft_descriptor] = "SOME DESCRIPTOR"
options[:express_request] = true
```
But I do not see descriptor correctly set in the credit card statement.

Going through the source file, I see this call stack:
`PAYPAL_GATEWAY.reference_transaction(price, options)` calls:
`build_reference_transaction_request('Sale', money, options)` calls
`add_payment_details(xml, money, currency_code, options)`  in `paypal_common_api`
Then inside this function, it calls:
`add_express_only_payment_details(xml, options) if options[:express_request]`
That's the reason I pass `options[:express_request] = true`
Then inside `add_express_only_payment_details` it sets up `soft_descriptor`.

So from my understanding, this all make senses. Not sure why descriptor does not show in bank statement.
